app/controllers/KairosDbProxyController.java

```diff
@@ -157,10 +157,15 @@ public KairosDbProxyController(
      */
     public CompletionStage<Result> queryMetrics() {
         try {
-        final MetricsQuery metricsQuery = _mapper.treeToValue(request().body().asJson(), MetricsQuery.class);
-        return _kairosService.queryMetrics(metricsQuery)
-                .<JsonNode>thenApply(_mapper::valueToTree)
-                .thenApply(Results::ok);
+            final MetricsQuery metricsQuery = _mapper.treeToValue(request().body().asJson(), MetricsQuery.class);
+            if (!metricsQuery.getMetrics().stream().allMatch(metric -> metric.getAggregators().size() >= 1)) {
```

Comment:
I would invert this condition to say `there is a metric whose aggregator list is empty` i.e.

```java
metricsQuery.getMetrics().stream().anyMatch(metric -> metric.getAggregators().isEmpty())
```
I think it makes intent clearer by removing a `!` and the comparison